GoogleDataTransport/GDTCORTests/Unit/GDTCORPlatformTest.m

```diff
@@ -0,0 +1,78 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <GoogleDataTransport/GDTCORPlatform.h>
+#import <GoogleDataTransport/GDTCORReachability.h>
+
+#import "GDTCORTests/Unit/GDTCORTestCase.h"
+
+@interface GDTCORPlatformTest : GDTCORTestCase
+
+@end
+
+@implementation GDTCORPlatformTest
+
+/** Tests the reachability of mobile network connection in current platform. */
+- (void)testMobileConnectionReachability {
+  SCNetworkReachabilityFlags reachabilityFlags = [GDTCORReachability currentFlags];
+  // The mobile network connection should be always false in simulator logic test.
+  XCTAssertFalse(GDTCORReachabilityFlagsContainWWAN(reachabilityFlags));
+}
+
+/** Tests network connection type message generating in current platform. */
+- (void)testGetNetworkConnectionType {
+  NSInteger networkConnectionType;
+  XCTAssertNoThrow(networkConnectionType = GDTCORNetworkTypeMessage());
+  // The network connection type should be always WIFI in simulator logic test.
+  XCTAssertEqual(networkConnectionType, GDTCORNetworkTypeWIFI);
+}
+
+/** Tests mobile network connection subtype generating in current platform. */
+- (void)testGetNetworkMobileSubtype {
+  NSInteger networkMobileSubtype;
+  XCTAssertNoThrow(networkMobileSubtype = GDTCORNetworkMobileSubTypeMessage());
+  // The network connection type should be always UNKNOWN in simulator logic test.
+  XCTAssertEqual(networkMobileSubtype, GDTCORNetworkMobileSubtypeUNKNOWN);
+}
+
+/** Tests the designated initializer of GDTCORApplication. */
+- (void)testInitializeGDTCORApplication {
+  GDTCORApplication *application;
+  XCTAssertNoThrow(application = [[GDTCORApplication alloc] init]);
+  XCTAssertNotNil(application);
+  XCTAssertFalse(application.isRunningInBackground);
+}
+
+/** Tests the sharedApplication generating of GDTCORApplication. */
+- (void)testGenerateSharedGDTCORApplication {
+  GDTCORApplication *application;
+  XCTAssertNoThrow(application = [GDTCORApplication sharedApplication]);
+  XCTAssertNotNil(application);
+}
+
+/** Tests background task creating of GDTCORApplication. */
+- (void)testGDTCORApplicationBeginBackgroundTask {
+  GDTCORApplication *application;
+  application = [[GDTCORApplication alloc] init];
+  __block GDTCORBackgroundIdentifier bgID;
+  XCTAssertNoThrow(bgID = [application beginBackgroundTaskWithName:@"GDTCORPlatformTest"
+                                                 expirationHandler:^{
+                                                   [application endBackgroundTask:bgID];
+                                                   bgID = GDTCORBackgroundIdentifierInvalid;
+                                                 }]);
```

Comment:
Make sure to add an equivalent endBackgroundTask call for bgID, wrap that in a NoThrow